Add support for XP-Pen Artist 22 Pro.

talanis85 opened the pull request:
Configuration for this tablet: https://www.xp-pen.com/product/artist-22-pro.html

Tested it for a while now on Linux. Everything seems to work. Did not test on other OSes.

jamesbt365 commented:
Diagnostic information is mandatory as per the [CONTRIBUTING.md](https://github.com/OpenTabletDriver/OpenTabletDriver/blob/master/CONTRIBUTING.md#adding-a-configuration). Go to help > export diagnostics and save that file and send it here.

jamesbt365 commented on OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 22 Pro.json:
```diff
@@ -0,0 +1,41 @@
+      "MaxX": 47663.0,
+      "MaxY": 26809.0
+    },
+    "Pen": {
+      "MaxPressure": 8191,
+      "Buttons": {
+        "ButtonCount": 2
+      }
+    },
```
This button format is incorrect for the master branch.

jamesbt365 commented on OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 22 Pro.json:
```diff
@@ -0,0 +1,41 @@
+      "InitializationStrings": []
+    }
+  ],
+  "AuxilaryDeviceIdentifiers": [],
```
The master branch fixes this typo.
```suggestion
  "AuxiliaryDeviceIdentifiers": [],
```

jamesbt365 commented on OpenTabletDriver.Configurations/Configurations/XP-Pen/Artist 22 Pro.json:
```diff
@@ -0,0 +1,41 @@
+        "ButtonCount": 2
+      }
+    },
+    "AuxiliaryButtons": {
+      "ButtonCount": 0
+    },
```
This should be null, not zero.

jamesbt365 reviewed (commented):
Alongside the review comments, you need to remove all null/empty fields to pass the tests.

jamesbt365 commented:
> Diagnostic information is mandatory as per the [CONTRIBUTING.md](https://github.com/OpenTabletDriver/OpenTabletDriver/blob/master/CONTRIBUTING.md#adding-a-configuration). Go to help > export diagnostics and save that file and send it here.

Just to let you know we are unable to review your PR properly until we get this information.

jamesbt365 commented:
Why did you change the base to `0.6.x`? Master was fine.

talanis85 commented:
Sorry. I should have based this off 0.6.x, not master. I changed the branch now and removed the empty fields in the configuration.

Diagnostics file is here: 
[diagnostics.json](https://github.com/user-attachments/files/18137374/diagnostics.json)

talanis85 commented:
> Why did you change the base to `0.6.x`? Master was fine.

The thing is, master does not work for me. So I cannot test it.

jamesbt365 commented:
Based off the lengths, are you sure you're using the right interface? It seems more likely that the correct one will be 10 (input) 8 (output). 64 is still a valid pick if it has the correct data on, but just wan to make sure that this actually works properly and isn't some quirk.

talanis85 commented:
Definitely not sure there. I based this off of an existing configuration of a different XP-Pen tablet with similar specs but I dont remember which one. It was just some trial and error.

With input 10 output 8 part of the pressure value seems to get interpreted as tilt (which i'm pretty sure this tablet/stylus does not support). But of course, I dont really understand what these lengths mean :)

According to the raw data coming from /dev/hidraw* the incoming records indeed seem to be 10 bytes each. I'm guessing that's what InputReportLength means?

talanis85 commented:
Upon further investigation, I noticed the following:

* Setting OutputReportLength to >0 causes the tablet to emit slightly different reports (causing the invalid tilt value). I'm assuming this is because of some faulty initialization report sent to the tablet. I can reproduce this by `echo -n ArAE > /dev/hidraw*` and then looking at the raw output from the device. Is it true that setting OutputReportLength to 0 means that the OutputInitReport is never sent?
* With OutputReportLength=0 and InputReportLength=10 (which should be the correct value), OTD correctly detects the tables but seems to attach to the wrong hidraw device. With InputReportLength=64 the correct device gets used.

mrdev023 commented:
Up

gonX commented:
Looking for further testers of this configuration, because at a glance it looks serviceable.
